scripts/test-cli.sh

```diff
@@ -48,11 +48,27 @@ test -s out || fail "scalajsp bin/Foo$.sjsir: empty output"
 scalajsp bin/Foo\$A.sjsir > out
 test -s out || fail "scalajsp bin/Foo\$A.sjsir: empty output"
 
-scalajsld -s -o test.js -mm Foo.main bin
+scalajsld -s -o test.js -mm Foo.main bin 2> test_stderr.txt
+grep -Fxq "Warning: using file as output is deprecated since 1.3.0. Please provide directory" test_stderr.txt \
+  || fail "expected warning. Got: $(cat test_stderr.txt)"
 test -s test.js || fail "scalajsld: empty output"
 test -s test.js.map || fail "scalajsld: empty source map"
 
-node test.js > got.run
+node test.js > got-legacy.run
+cat > want-legacy.run <<EOF
+asdf 2
+EOF
+
+diff got-legacy.run want-legacy.run
+
+mkdir test-output
+scalajsld -s --outputDir test-output --moduleSplitStyle SmallestModules --moduleKind CommonJSModule -mm Foo.main bin
+test -s test-output/Foo.js || fail "scalajsld: empty output"
+test -s test-output/Foo.js.map || fail "scalajsld: empty output"
```

Comment:
These file names are not specified (or more precisely: the module ID is an implementation detail subject to change). Maybe just assert that there is more than one `.js` file?